src/display/annotation_layer.js

```diff
@@ -2049,7 +2049,8 @@ class AnnotationLayer {
    */
   static render(parameters) {
     const sortedAnnotations = [],
-      popupAnnotations = [];
+      popupAnnotations = [],
+      sameName = new Map();
```

Comment:
Could annotations with the same name ever be found on *different* pages, because if so I'm not sure that this will work (since this is limited to just the current page)?

Reply:
That's correct.

Reply:
And I'd say that it was already an issue (I mean before this patch).

Reply:
My (possibly very stupid) idea when I initially saw the issue was that you maybe track the created `input` elements in e.g. a `WeakSet` defined "globally" in this file, and then just check the elements that we lookup with `document.getElementsByName`  against that.

Based on the following quote from https://github.com/mozilla/pdf.js/issues/14003#issue-993960056, Rob probably has better ideas though:
> I suggest to replace the document.getElementsByTagName calls with a function that checks that the elements are created by the annotation_layer.js code before returning them. There are several ways to do so, e.g. by adding a custom attribute to each input element.

Reply:
It won't avoid the problem with fields with same name across different pages: `document.getElementsByName` will get only the visible elements.

Reply:
> And I'd say that it was already an issue (I mean before this patch).

I see what you're saying, but it seems to me (I may be wrong of course) that it'd be slightly worse since before you got the Annotations on all active pages whereas with this patch it's *only* the current page.

---

Anyway, it seems that we might need to expand the `AnnotationStorage` to also have a `name -> id` mapping such that the initial value gets set correctly when rendering new pages? If so, that probably shouldn't be too hard to implement (famous last words).

Reply:
It quite ugly, and possibly not correct in all cases, but the following patch is something that I just threw together quickly in order to try tracking both the `id` and `name` in the AnnotationStorage: https://github.com/mozilla/pdf.js/compare/master...Snuffleupagus:AnnotationStorage-name

Reply:
When I filed the issue I was thinking of something simple like using a `WeakSet`, tagging the element with a custom attribute, or even just prepending a prefix to the name to have something unique that can't overlap with existing ones. This draft patch looks more complex than that - why?

Reply:
@calixteman Sorry if I'm massively overstepping here, but I've got a (possibly stupid) idea.

What if we utilized the data returned by the `getFieldObjects`-method in the API, since then we'd be able to both access the fields on all active pages *and* also update the AnnotationStorage for not-yet-rendered pages?

https://github.com/mozilla/pdf.js/compare/master...Snuffleupagus:AnnotationLayer-getElementsByName

Reply:
@Snuffleupagus, no, you aren't and I think that's a very good idea.
This way we don't have to change `AnnotationStorage` implementation because we'll have all the ids for fields with the same name. Would you mind to write the patch since you've almost thought about everything here ? If yes, please close this one.

@Rob--W, my idea was to remove the call to `getElementsByName` and just update the fields with the same name. But as pointed out by Jonas, it was a bad idea and in the meantime it was not so bad because it helped us to find a bug.